src/main/java/io/luna/net/msg/in/CastOnPlayerMessageReader.java

```diff
@@ -0,0 +1,36 @@
+package io.luna.net.msg.in;
+
+import io.luna.game.event.Event;
+import io.luna.game.event.impl.CastOnPlayerEvent;
+import io.luna.game.model.World;
+import io.luna.game.model.mob.Player;
+import io.luna.net.codec.ByteOrder;
+import io.luna.net.codec.ValueType;
+import io.luna.net.msg.GameMessage;
+import io.luna.net.msg.GameMessageReader;
+
+public final class CastOnPlayerMessageReader extends GameMessageReader {
+
+    @Override
+    public Event read(Player player, GameMessage msg) {
+        int targetPlayerId = msg.getPayload().getShort(false, ValueType.ADD, ByteOrder.LITTLE); // the id of the player
+        // targeted
+        int targetSpellId = msg.getPayload().getShort(false, ValueType.ADD); // the id of the spell casted
```

Comment:
These values have not been tested yet. If anyone would like to help out, I can point you to the corresponding client code.

Reply:
Some code from my previous release says its 
playerIndex - unsigned short, value type A
spellId - unsigned short, byte order little